UefiPayloadPkg/Library/BuildFdtLib/X86_BuildFdtLib.c

```diff
@@ -0,0 +1,976 @@
+                                              PCI_DEVICE_ID_OFFSET,
+                                              sizeof (PciData.Hdr.DeviceId),
+                                              &(PciData.Hdr.DeviceId)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint8,
+                                              PCI_REVISION_ID_OFFSET,
+                                              sizeof (PciData.Hdr.RevisionID),
+                                              &(PciData.Hdr.RevisionID)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_SVID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemVendorID),
+                                              &(PciData.Device.SubsystemVendorID)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_SID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemID),
+                                              &(PciData.Device.SubsystemID)
+                                              );
+        }
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemVendorID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-vendor-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.RevisionID);
+        Status = FdtSetProp (Fdt, GmaNode, "revision-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.DeviceId);
+        Status = FdtSetProp (Fdt, GmaNode, "device-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.VendorId);
+        Status = FdtSetProp (Fdt, GmaNode, "vendor-id", &Data16, sizeof (UINT16));
+      }
+
+      if (SerialParent != NULL) {
+        DEBUG ((DEBUG_INFO, "SerialParent->IsIsaCompatible  :%x , SerialParent->ParentDevicePcieBaseAddress :%x\n", SerialParent->IsIsaCompatible, SerialParent->ParentDevicePcieBaseAddress));
+        DEBUG ((DEBUG_INFO, "BusBase  :%x , PciRootBridgeInfo->RootBridge[Index].Bus.Base :%x\n", BusBase, PciRootBridgeInfo->RootBridge[Index].Bus.Base));
+      }
+
+      if (BusBase == PciRootBridgeInfo->RootBridge[Index].Bus.Base) {

```

Comment:
check if BusBase fall within the range:
if (BusBase >= ....Bus.Base && BusBase <= ...Bus.Limit)

Reply:
resolved